roles/lib_openshift/library/docker_creds.py

```diff
@@ -0,0 +1,207 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+DOCUMENTATION = '''
+---
+module: docker_creds
+
+short_description: Creates/updates a 'docker login' file in place of using 'docker login'
+
+version_added: "2.4"
+
+description:
+    - This module creates a docker config.json file in the directory provided by 'path'
+      on hosts that do not support 'docker login' but need the file present for
+      registry authentication purposes of various other services.
+
+options:
+    path:
+        description:
+            - This is the message to send to the sample module
+        required: true
+    registry:
+        description:
+            - This is the registry the credentials are for.
+        required: true
+    username:
+        description:
+            - This is the username to authenticate to the registry with.
+        required: true
+    password:
+        description:
+            - This is the password to authenticate to the registry with.
+        required: true
+
+author:
+    - "Michael Gugino <mgugino@redhat.com>"
+'''
+
+EXAMPLES = '''
+# Pass in a message
+- name: Place credentials in file
+  docker_creds:
+    path: /root/.docker
+    registry: registry.example.com:443
+    username: myuser
+    password: mypassword
+'''
+
+import base64
+import json
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+def check_dest_dir_exists(module, dest):
+    '''Check if dest dir is present and is a directory'''
+    dir_exists = os.path.exists(dest)
+    if dir_exists:
+        if not os.path.isdir(dest):
+            msg = "{} exists but is not a directory".format(dest)
+            result = {'failed': True,
+                    'changed': False,
+                    'msg': msg,
+                    'state': 'unknown'}
+            module.fail_json(**result)
+        else:
+            return 1
+    else:
+        return 0
+
+def create_dest_dir(module, dest):
+    try:
+        os.makedirs(dest, mode=0o700)
+    except Exception as e:
+        result = {'failed': True,
+                'changed': False,
+                'msg': str(e),
+                'state': 'unknown'}
+        module.fail_json(**result)
+
+def load_config_file(module, dest):
+    '''load the config.json in directory dest'''
+    conf_file_path = os.path.join(dest, 'config.json')
+    if os.path.exists(conf_file_path):
+        # Try to open the file
+        try:
+            with open(conf_file_path) as f:
+                data = f.read()
+        # Couldn't open the file, exit.
+        except Exception as e:
+            result = {'failed': True,
+                    'changed': False,
+                    'msg': str(e),
+                    'state': 'unknown'}
+            module.fail_json(**result)
+        try:
+            j = json.loads(data)
+        # Invalid json data.
+        except Exception as e:
```

Comment:
Could this try/except block be merged into the one above?  Seems like the exception handling is identical.

Reply:
Good suggestion.  Since I'm catching all the errors and just dumping the string, probably can combine these.